File: app/src/main/java/com/youtility/repository/base/UserRepository.java
package com.youtility.repository.base;

import io.reactivex.Observable;

/**
 * Created by Mohit Rajput on 20/3/19.
 * Current users profile data provider
 */
public interface UserRepository {
    public Observable<Boolean> isUserLoggedIn();

    public Observable<String> getFirstName();

    public void saveFirstName(String firstName);

    public void saveLastName(String lastName);

    public Observable<String> getLastName();
}

File: energyswitchcui/src/main/kotlin/com/youtility/energyswitchcui/repository/base/UserRepository.kt
package com.youtility.energyswitchcui.repository.base

import com.youtility.energyswitchcui.model.AuthToken
import io.reactivex.Observable

/**
 * Created by Mohit Rajput on 2/4/19.
 * Current users profile data provider
 */
interface UserRepository {
    fun getAuthDetails(mobileNumber: String, passcode: String, clientId: String, grantType: String, scope: String): Observable<AuthToken>

    fun saveAuthDetails(authToken: AuthToken)
}